mne/simulation/source.py

```diff
@@ -218,3 +220,126 @@ def simulate_stc(src, labels, stc_data, tmin, tstep, value_fun=None):
 
     stc = SourceEstimate(data, vertices=vertno, tmin=tmin, tstep=tstep)
     return stc
+
+
+def simulate_stc_from_dipole(dipole, bem, info, trans=None, min_dist=5.0,
+                             n_jobs=1, verbose=None):
+    """Convert dipole object to source estimate and calculate forward operator
+
+    The instance of Dipole is converted to a discrete volume source space,
+    which is then combined with a BEM (can also be a sphere model dict) and
+    the sensor information in info to form a forward operator.
+
+    The source estimate object (with the forward operator) can be projected to
+    sensor-space using :func:`mne.simulation.evoked.simulate_evoked`.
+
+    Parameters
+    ----------
+    dipole : Dipole object
+        Instance of Dipole containing position, orientation and amplitude of
+        one or more dipoles. Several dipoles may occur simultaneously, but
+        when more than one time point is defined, the spacing between them
+        (i.e. the sampling rate) must be constant.
+    bem : str | dict
+        The BEM filename (str) or a loaded sphere model (dict).
+    info : dict
+        Sensor-info etc., e.g., from real evoked file.
+    trans : str | None
+        The head<->MRI transform filename. Must be provided unless BEM
+        is a sphere model.
+    min_dist : float
+        Minimum distance (in milimeters) from the dipole to the inner skull.
+    # model_meg : bool
+    #     Include MEG channels (with or without compensation) in model.
+    # model_eeg : bool
+    #     Include EEG channels in model.
+    n_jobs : int
+        Number of jobs to run in parallel (used in making forward solution).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    stc : instance of VolSourceEstimate
+        The dipoles converted to a discrete set of points and associated
+        time courses.
+    fwd : instance of Forward
+        The forward solution.
+
+    See Also
+    --------
+    mne.simulation.simulate_evoked
+
+    Notes
+    -----
+    .. versionadded:: 0.12.0
+    """
+    # Make copies to avoid mangling original dipole
+    tims = dipole.times.copy()
+    poss = dipole.pos.copy()
+    amps = dipole.amplitude.copy()
+    oris = dipole.ori.copy()
```

Comment:
please don't be creative with variable names.

call them as the attributes : times, pos, amplitude, ori